### PR TITLE
NME Fixes

### PR DESCRIPTION
**Pull request by msDestiny14**
-Fix for inputLineComponents
-Fix for popup not appearing

### CODE REVIEW

File: nodeEditor/src/blockTools.ts [hunk 1]
```diff
@@ -124,7 +124,7 @@ export class BlockTools {
                 return new TextureBlock("Texture", nodeMaterial.mode === NodeMaterialModes.Particle);
             case "ReflectionTextureBlock":
                 return new ReflectionTextureBlock("Reflection texture");
-            case "LightBlock":
+            case "LightsBlock":
```
Comment (deltakosh) [inline nodeEditor/src/blockTools.ts:127]: why this change?

File: nodeEditor/src/sharedComponents/floatLineComponent.tsx [hunk 2]
```diff
@@ -58,6 +58,15 @@ export class FloatLineComponent extends React.Component<IFloatLineComponentProps
         return false;
     }
 
+    componentWillUnmount() {
+        if (this.props.globalState.blockKeyboardEvents = true) {
```
Comment (deltakosh) [inline nodeEditor/src/sharedComponents/floatLineComponent.tsx:62]: ===

File: nodeEditor/src/sharedComponents/numericInputComponent.tsx [hunk 3]
```diff
@@ -35,6 +35,12 @@ export class NumericInputComponent extends React.Component<INumericInputComponen
         return false;
     }
 
+    componentWillUnmount() {
+        if (this.props.globalState.blockKeyboardEvents = true) {
```
Comment (deltakosh) [inline nodeEditor/src/sharedComponents/numericInputComponent.tsx:39]: ===

File: nodeEditor/src/blockTools.ts (hunk 1 above)
Comment (msDestiny14) [reply nodeEditor/src/blockTools.ts:127]: @PirateJC asked me to change it.

File: nodeEditor/src/sharedComponents/textInputLineComponent.tsx [hunk 4]
```diff
@@ -38,6 +38,13 @@ export class TextInputLineComponent extends React.Component<ITextInputLineCompon
         return false;
     }
 
+    componentWillUnmount() {
+        if (this.props.globalState.blockKeyboardEvents = true) {
```
Comment (deltakosh) [inline nodeEditor/src/sharedComponents/textInputLineComponent.tsx:42]: ===

File: nodeEditor/src/sharedComponents/textInputLineComponent.tsx [hunk 5]
```diff
@@ -61,7 +68,7 @@ export class TextInputLineComponent extends React.Component<ITextInputLineCompon
         this._localChange = true;
         const store = this.props.value !== undefined ? this.props.value : this.props.target[this.props.propertyName!];
 
-        if (this.props.validator && raisePropertyChanged) {
+        if (this.props.validator) {
```
Comment (deltakosh) [inline nodeEditor/src/sharedComponents/textInputLineComponent.tsx:71]: why removing raisePropertyChanged?

**Review by deltakosh — CHANGES_REQUESTED** [review]
.

File: nodeEditor/src/sharedComponents/textInputLineComponent.tsx (hunk 5 above)
Comment (msDestiny14) [reply nodeEditor/src/sharedComponents/textInputLineComponent.tsx:71]: to get the pop up showing again. I was actually wondering if I should remove the lines a bit lower that you added back in july since it will now do the same thing.

**Comment by msDestiny14** [comment]
Updated comment to better overview the changes.

File: nodeEditor/src/blockTools.ts (hunk 1 above)
Comment (deltakosh) [reply nodeEditor/src/blockTools.ts:127]: This could be misleading when people will found out that the block name is not the same

File: nodeEditor/src/sharedComponents/textInputLineComponent.tsx (hunk 5 above)
Comment (deltakosh) [reply nodeEditor/src/sharedComponents/textInputLineComponent.tsx:71]: actually this should work

File: nodeEditor/src/blockTools.ts (hunk 1 above)
Comment (Popov72) [reply nodeEditor/src/blockTools.ts:127]: More importantly it's a breaking change, all node materials that are currently using `LightBlock` will fail loading.

File: nodeEditor/src/sharedComponents/textInputLineComponent.tsx (hunk 5 above)
Comment (deltakosh) [reply nodeEditor/src/sharedComponents/textInputLineComponent.tsx:71]: actually no: we do not want to validate on every change (only on blur) else we will prevent users to enter values (like the validator is that you cannot enter "var" but you want to type "variant". If you validate per stroke you will never be able to enter variant

File: nodeEditor/src/blockTools.ts (hunk 1 above)
Comment (deltakosh) [reply nodeEditor/src/blockTools.ts:127]: Totally. I should have mentioned that also. I'm getting old

Comment (msDestiny14) [reply nodeEditor/src/blockTools.ts:127]: OH. That stinks I will remove it then.

File: nodeEditor/src/sharedComponents/textInputLineComponent.tsx (hunk 5 above)
Comment (msDestiny14) [reply nodeEditor/src/sharedComponents/textInputLineComponent.tsx:71]: the code you have below does similar should i remove that segment as well then?

Comment (deltakosh) [reply nodeEditor/src/sharedComponents/textInputLineComponent.tsx:71]: I did not check but my ubber point is to make sure that your fix works with a variable that you want to name uv1 for instance. The system should block uv but not uv1

Comment (msDestiny14) [reply nodeEditor/src/sharedComponents/textInputLineComponent.tsx]: Changed :)

**Review by deltakosh — APPROVED** [review]
